qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -425,26 +459,44 @@ def _synth_natural_direction(
         if (
             natural_direction in {None, True}
             and preferred_direction is None
-            and gate_lengths
-            and gate_errors
+            and ((gate_lengths and gate_errors) or target)
         ):
             len_0_1 = inf
             len_1_0 = inf
-            twoq_gate_lengths = gate_lengths.get(decomposer2q.gate.name)
-            if twoq_gate_lengths:
-                len_0_1 = twoq_gate_lengths.get((qubits[0], qubits[1]), inf)
-                len_1_0 = twoq_gate_lengths.get((qubits[1], qubits[0]), inf)
-                if len_0_1 < len_1_0:
-                    preferred_direction = [0, 1]
-                elif len_1_0 < len_0_1:
-                    preferred_direction = [1, 0]
-                if preferred_direction:
+            twoq_gate_props = None
+            if target is not None:
```

Comment:
Would it be simpler to just populate `gate_lengths` and `gate_errors` from target if they are defined to avoid these similar looking conditionals?

Reply:
I was thinking we would want to pass the target as the source of truth to plugins and have it supersede any of the other options payloads if it's available. My concern is mostly that we need the target to do the non-global 2q piece above and when I wrote this I was mostly concerned about the target not being in sync with the `gate_lengths` and `gate_errors` payloads. But I ended up having to modify the `_build_gate_errors()` and `_build_gate_lengths()` functions if a target is available so maybe it's safe if we drop this here and only use the target for the coupling map part.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7775/commits/771cfbacb1f68e65b3838c7658ad2d686ab2a93c